services/com.mde-network.ep.toolfunctions.xtext/src/main/java/com/mdenetnetwork/ep/toolfunctions/xtext/XtextTool.java

```diff
@@ -69,18 +69,32 @@ public void run(String languageName, String baseName, String extension, String g
 		
 		CliProjectsCreatorMain.main(args);
 		
-		
-		
-		
 		// Add activity files
-		if (grammar != null) {
-			final String grammarSrcPath = languageName.replace('.', '/');
-			final String xtextGrammarPath= PROJECT_PATH + baseName + "/src/" + grammarSrcPath  + ".xtext";
-			recreateGrammarFile(xtextGrammarPath, grammar);
+        //     TODO support multiple file types
+        final String grammarPath = PROJECT_PATH + baseName + "/src/" + baseName.replace('.', '/');
+        final String projectSrcPath = Path.of(grammarPath).getParent().toString();
+        
+        final String shortName = languageName.replace(baseName + ".", "");
+
+		if (grammar != null && !grammar.equals("undefined") ) {
+			final String xtextGrammarPath= grammarPath + "/" + shortName + ".xtext";
+			recreateProjectFile(xtextGrammarPath, grammar);
+		}
+
+		if (validator != null && !validator.equals("undefined") ) {
+			final String xtextValidatorPath= projectSrcPath + "/validation/" + shortName + "Validator.xtend";
```

Comment:
Shouldn't this be `grammarPath + "/validation/" + shortName + "Validator.xtend"`? The validator usually lives in a sub-directory of the directory containing the `.xtext` file.

Reply:
See above to address in https://github.com/mdenet/platformtools/issues/51